exchange/exchange.go

```diff
@@ -70,13 +70,13 @@ func New(ctx context.Context, h host.Host, ds datastore.Batching, opts Options)
 		return nil, err
 	}
 
-	// Make a new default key to be sure we have an address where to receive our payments
 	if opts.Wallet.DefaultAddress() == address.Undef {
-		_, err = opts.Wallet.NewKey(ctx, wallet.KTSecp256k1)
+		_, err = utils.NewKey(ctx, opts.Wallet)
```

Comment:
I had to keep a place in the exchange where we create an address if none was created before.
It means that for the usual use of Pop, the node will alway take care of the creation/importation of keys, but for the tests, when we don't instantiate any node, they will fail.
The problem is that we can now create keys at 2 different places (node & exchange), and we can give different key type argument to our NEwKey function. To prevent having having inconsistencies between our tests & real world, I created a global NewKey function that already encapsulate the type of key we chose for the whole project.

Reply:
@gallexis I think the key type doesn't matter, the tests and behavior will be the same for any key type used. I don't think we need that global NewKey function.